src/tokenize.c

```diff
@@ -54,24 +59,109 @@ static char *DefaultNormalize(char *s, char *dst, size_t *len) {
   int escaped = 0;
   for (size_t ii = 0; ii < origLen; ++ii) {
     if (isupper(s[ii])) {
+      // convert the character to lower case
       SWITCH_DEST();
       realDest[dstLen++] = tolower(s[ii]);
     } else if ((isblank(s[ii]) && !escaped) || iscntrl(s[ii])) {
+      // skip blank characters and control characters
       SWITCH_DEST();
     } else if (s[ii] == '\\' && !escaped) {
+      // handle backslash escapes
       SWITCH_DEST();
       escaped = 1;
       continue;
     } else {
+      // copy the character as is
       dst[dstLen++] = s[ii];
     }
+    // reset the escape flag
+    escaped = 0;
+  }
+
+  *len = dstLen;
+  return dst;
+}
+
+
+/**
+ * Normalize a UTF-8 string, converting it to lower case and removing control 
+ * characters.
+ * - s contains the raw token
+ * - dst is the destination buffer which contains the normalized text
+ * - len on input contains the length of the raw token. On output contains the
+ *   length of the normalized token
+ */
+static char *DefaultNormalize_utf8(char *s, char *dst, size_t *len) {
+  // TODO: set locale depending on the index language?
+  setlocale(LC_ALL, "en_US.UTF-8");
```

Comment:
should we reset the local upon exiting the function?
This is going to affect the entire process, no?
Might also propagate to redis and other modules.